cpp/include/raft/core/device_mdspan.hpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/detail/accessor_mixin.hpp>
+#include <raft/core/mdspan.hpp>
+
+namespace raft {
+
+template <typename AccessorPolicy>
+using device_accessor = detail::accessor_mixin<AccessorPolicy, false, true>;
+
+template <typename AccessorPolicy>
+using managed_accessor = detail::accessor_mixin<AccessorPolicy, true, true>;
```

Comment:
`accessor_mixin<AccessorPolicy, true, true>` just implies that the memory is accessible from either host or device.  It could be either pinned memory or managed memory.  Should we thus consider a different name for this alias, such as `host_device_accessor`?